### PR TITLE
chore(deps): update golangci actions

### PR DESCRIPTION
**Pull request by karasz**
## Summary by CodeRabbit

- **New Features**
	- Expanded Go version testing in the build process to include versions 1.21, 1.22, and 1.23.
  
- **Bug Fixes**
	- Improved conditional logic for the lint job to prevent redundant checks on pull requests from the same repository.

- **Chores**
	- Upgraded the Go setup action to version 5 across all workflows.
	- Updated the Go version in the workflows to 1.21.
	- Updated the GoReleaser action to version 6 in the release workflow.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -14,10 +14,10 @@ jobs:
     if: (github.event_name == 'push' || github.event.pull_request.head.repo.full_name != github.repository)
     strategy:
       matrix:
-        go: [ '1.20', '1.21' ]
+        go: [ '1.21', '1.22', '1.23' ]
```
[the remark this answers is not in the document]
Comment (karasz) [reply .github/workflows/build.yml:17]: latest stable go version is 1.23.4 (we are in December 2024 so February 2024 is gone already)

[the remark this answers is not in the document]
Comment (karasz) [reply .github/workflows/build.yml:17]: I pushed a fix in commit f360cff77d0c19e2a28f388443aa8b6a53f4a0de, please review it.

**Comment by karasz** [comment]
I pushed a fix in commit https://github.com/darvaza-proxy/kube/commit/f360cff77d0c19e2a28f388443aa8b6a53f4a0de, please review it.

**Comment by karasz** [comment]
@coderabbitai full review

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -15,7 +15,7 @@ jobs:
       - name: Fetch all tags
         run: git fetch --force --tags
       - name: Set up Go
-        uses: actions/setup-go@v4
+        uses: actions/setup-go@v5
         with:
           go-version: '~1.20'
```
[the remark this answers is not in the document]
Comment (karasz) [reply .github/workflows/release.yml]: fixed

**Comment by karasz** [comment]
@coderabbitai full review

File: .github/workflows/lint.yml [hunk 3]
```diff
@@ -15,9 +15,9 @@ jobs:
     if: (github.event_name == 'push' || github.event.pull_request.head.repo.full_name != github.repository)
     steps:
       - name: Setup Go
-        uses: actions/setup-go@v4
+        uses: actions/setup-go@v5
         with:
-          go-version: '~1.20'
+          go-version: '~1.22'
```
[the remark this answers is not in the document]
Comment (karasz) [reply .github/workflows/lint.yml]: fixed

**Review by amery — APPROVED** [review]

**Pull request merged**